lib/ASRouter.jsm

```diff
@@ -49,6 +49,16 @@ const STARTPAGE_VERSION = "6";
 const MessageLoaderUtils = {
   STARTPAGE_VERSION,
   REMOTE_LOADER_CACHE_KEY: "RemoteLoaderCache",
+  _errors: [],
+
+  reportError(e) {
+    Cu.reportError(e);
+    this._errors = [{timestamp: new Date(), error: e}, ...this._errors];
+  },
+
+  get errors() {
```

Comment:
I like the `reportError` strategy, but personally I would return the error through `loadMessagesForProvider` so the state can be centralized in `ASRouter._state`, or maybe even as a property inside provider in `ASRouter._state.providers` itself?

Keeping it here makes cleanup more complicated IMO